src/ploomber_core/telemetry/telemetry.py

```diff
@@ -682,7 +691,19 @@ def wrapper(*args, **kwargs):
 
                 try:
                     if payload:
-                        result = func(_payload, *args, **kwargs)
+                        if is_method:
+                            """
+                            If the method is defined in Class, the original parameter:
+                            def method(self, parameter1, parameter2,...)
+                            We will inject the payload parameter right after the self
+                            To become:
+                            def method(self, payload, parameter1, parameter2,...)
+                            """
```

Comment:
let's make this a comment instead of a string. also try to shorten it

Reply:
also, don't forget to add a changelog entry

Reply:
Added in [fe659ab](https://github.com/ploomber/core/pull/45/commits/fe659ab09c03bf507dd5cd4e218c8b857cd650f5)